### ISSUE TITLE
Training crash

### ISSUE THREAD

**Issue by 337188754** [Mengqi-Lei/ConDSeg#3]
Hello, Creator,

I was able to successfully reproduce the results of this model on the Kvasir-SEG dataset. However, when I ran it on a retinal vessel dataset, the training crashed. I would like to ask if there are any other modifications I should make besides adjusting the hyperparameters. Additionally, I was wondering if there might be a missing piece of code for loading checkpoints in the training script.

I look forward to your response. Thank you!

**Comment by Mengqi-Lei** [comment]
Thank you for reaching out with your questions.
Could you please clarify at which stage the training crashed? Specifically, was the training interrupted due to any particular reason, or did the model fail to converge?

**Comment by 337188754** [comment]
Thank you for your patient response. The model has indeed not converged. During the first phase of training, as the number of epochs increased, the metrics continuously declined until reaching zero, while the prediction metrics rose to one. It seems that the model has only learned the background, which I find very puzzling. I have tried adjusting the learning rate and loss function, but the issue remains unresolved. I would appreciate your guidance on this matter.
Epoch: 19 | Epoch Time: 1m 12s
	Train Loss: 3.2317 - mIoU: 0.0600 - F1: 0.1105 - Recall: 0.0770 - Precision: 0.4720
	 Val. Loss: 1.5645 - mIoU: 0.0029 - F1: 0.0058 - Recall: 0.0030 - Precision: 0.8985

Epoch: 20 | Epoch Time: 1m 11s
	Train Loss: 3.2103 - mIoU: 0.0514 - F1: 0.0955 - Recall: 0.0635 - Precision: 0.4883
	 Val. Loss: 1.5631 - mIoU: 0.0011 - F1: 0.0021 - Recall: 0.0011 - Precision: 0.9643

Epoch: 21 | Epoch Time: 1m 14s
	Train Loss: 3.1970 - mIoU: 0.0432 - F1: 0.0807 - Recall: 0.0514 - Precision: 0.5183
	 Val. Loss: 1.5611 - mIoU: 0.0000 - F1: 0.0000 - Recall: 0.0000 - Precision: 1.0000

Epoch: 22 | Epoch Time: 1m 11s
	Train Loss: 3.1844 - mIoU: 0.0394 - F1: 0.0739 - Recall: 0.0463 - Precision: 0.5542
	 Val. Loss: 1.5604 - mIoU: 0.0000 - F1: 0.0000 - Recall: 0.0000 - Precision: 1.0000

**Comment by Mengqi-Lei** [comment]
You mentioned that you were able to reproduce the results on the kvasir-seg dataset, but encountered a crash when training on the retinal dataset. I believe this could be due to either an issue with the code when migrating the new dataset, or a problem with the dataset itself (for example, an insufficient number of training samples, which might prevent the model from learning general pathological features and patterns.
We recommend that you first check the code related to the new dataset to ensure there are no errors. If the performance is still unsatisfactory, we suggest trying to skip the initial training phase and start directly from stage 2, then observe whether the expected performance can be achieved.

**Comment by 337188754** [comment]
Indeed, after skipping the first stage of training and directly entering the second stage, the metrics returned to a normal range. I will continue to explore the differences between the two stages in the code. Thanks for your explanation.

**Comment by lyj107** [comment]
Hi, I'm sorry to bother you, I'd like to ask you what version of the library you used when reproducing the code for this paper, could you please let me know?

**Comment by 337188754** [comment]
@lyj107 My virtual environment was created when I ran other models previously, and it contains many unnecessary libraries. So I can only tell you the versions of a few important libraries. The Python version is 3.10.16, and the corresponding CUDA toolkit version for PyTorch is 12.1.      
albumentations                2.0.5
numpy                         2.2.3
pillow                        10.3.0
torch                         2.3.0
torchaudio                    2.3.0
torchvision                   0.18.0

**Comment by lyj107** [comment]
> 我的虚拟环境是在我之前运行其他模型时创建的，它包含许多不必要的库。所以我只能告诉你几个重要库的版本。Python 版本为 3.10.16，PyTorch 对应的 CUDA 工具包版本为 12.1。相册 2.0.5 Numpy 2.2.3 枕头 10.3.0 火炬 2.3.0 TorchAudio 2.3.0 火炬视觉 0.18.0

Thank you very much for your reply, the problem I'm currently experiencing is mainly with the version of tensorflow and keras, would you be so kind as to tell me what version of these two you're using? Looking forward to your reply, thanks.

**Comment by Treetreestack** [comment]
Hi, I'm sorry to bother you, I encountered the same issue as you mentioned — after training in the first stage, the precision keeps being 1.0 while all other metrics (mIoU, F1, Recall) remain 0. Have you managed to solve this problem? I would really appreciate any insights you could share.

**Comment by M-moon387** [comment]
@337188754 Hello, I'm sorry to bother you, but I've encountered the same issue. Even after skipping the first stage of training, it still doesn't work. I was wondering if you have managed to resolve this problem?

**Comment by wenchuanjiang** [comment]
> 你好，创造者，
> 
> 我已成功在 Kvasir-SEG 数据集上复现了该模型的结果。但是，当我在视网膜血管数据集上运行时，训练崩溃了。我想请问除了调整超参数之外，是否还有其他需要修改的地方？另外，我想知道训练脚本中是否缺少加载检查点的代码。
> 
> 期待您的回复。谢谢！

Hello, how are your reproduction results on Kvasir-SEG and ISIC-2017? Compared with the original paper, my results are about two percentage points lower.
